components/apimgt/org.wso2.carbon.apimgt.usage/org.wso2.carbon.apimgt.usage.client/src/main/java/org/wso2/carbon/apimgt/usage/client/impl/APIUsageStatisticsRdbmsClientImpl.java

```diff
@@ -1607,7 +1604,7 @@ public int compare(APIResponseTimeDTO o1, APIResponseTimeDTO o2) {
             public int compare(APIVersionLastAccessTimeDTO o1, APIVersionLastAccessTimeDTO o2) {
                 // Note that o2 appears before o1
                 // This is because we need to sort in the descending order
-                return o2.getLastAccessTime().compareToIgnoreCase(o1.getLastAccessTime());
+                return (int) (o2.getLastAccessTime() - o1.getLastAccessTime());
```

Comment:
You sure this can always be casted to int? What is the time unit here? If there is a possibility of this becoming a very large number casting to int will fail.